server/src/main/scala/com/cloudera/livy/utils/SparkYarnApp.scala

```diff
@@ -50,6 +58,47 @@ object SparkYarnApp extends Logging {
 
   private def getYarnPollInterval(livyConf: LivyConf): FiniteDuration =
     livyConf.getTimeAsMs(LivyConf.YARN_POLL_INTERVAL) milliseconds
+
+  private val appType = Set("SPARK").asJava
+
+  private val leakagedAppTags = new java.util.concurrent.ConcurrentHashMap[String, Long]()
```

Comment:
leakagedAppTags is not persisted across restart. We should persist it to the state store.